mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/model/spi/PojoTypeModel.java

```diff
@@ -18,5 +18,5 @@
 	 */
 	PojoRawTypeModel<? super T> getRawType();
 
-	PojoPropertyModel<?> getProperty(String propertyName);
```

Comment:
That will have to be a no from me.

What you're saying here is "this method can provide you with a `PojoPropertyModel<P>`, where P is whatever you want". Which is wrong; actually this method can provide callers with a `PojoPropertyModel<P>`, where `P` is some pre-defined type. Not "whatever they want".

So this method should actually return `PojoPropertyModel<?>`, and if callers need something else, they will have to handle it on the call site.

From what I can see, there's only one place where my suggestion would raise an issue: `org.hibernate.search.mapper.pojo.model.path.impl.BoundPojoModelPathTypeNode#property`. But even there, that's only a problem because you changed the signature from:

```
public BoundPojoModelPathPropertyNode<T, ?> property(PropertyHandle propertyHandle)
```

to:

```
public <P> BoundPojoModelPathPropertyNode<T, P> property(PropertyHandle<P> propertyHandle)
```

If you just use a wildcard instead of a generic, you should be fine:

```
public BoundPojoModelPathPropertyNode<T, ?> property(PropertyHandle<?> propertyHandle)
```

Callers do not need more anyway.

Reply:
In `BoundPojoModelPathTypeNode` you just reverted your changes. I gave you the correct code, which includes a `<?>` instead of leaving a raw type.

I'll do it.